.github/workflows/release.yml

```diff
@@ -322,13 +356,13 @@ jobs:
 
   windows:
     name: Build native Windows binary
-    needs: [build-lib-jar, build-cli-jar]
+    needs: [build-lib-jar, build-cli-jar-win]
```

Comment:
Do we really need to build a jar with windows just to use for windows graal build? a jar compiled on any env should be compatible with any JVM in any OS right?

Reply:
The final executable is behaving erratically when I create the uber jar on Linux and compile natively on windows. 

I've  traced it down to the following `babashka.fs/path-separator` constant, which ends up with the Linux value (`:`), instead of the windows one (`;`):

https://github.com/babashka/fs/blob/60aea913a99a39b430a3666b7f5d116463c640c4/src/babashka/fs.cljc#L239

``` clojure
(def ^:const path-separator File/pathSeparator)
```

I suppose this expected because the constant is set at when compiling to java and thus takes the value of the compiling platform rather than the runtime one?

https://github.com/clojure/clojure/blob/master/changes.md

```
^:const lets you name primitive values with speedier reference.

    (def constants
     {:pi 3.14
      :e 2.71})

    (def ^:const pi (:pi constants))
    (def ^:const e (:e constants))

The overhead of looking up :e and :pi in the map happens at compile time, as (:pi constants) and (:e constants) are evaluated when their parent def forms are evaluated.
```

This has the bad effect that when `babashka.fs` is trying to split the `PATH` env variable on windows, it uses the wrong path separator and as such can't locate executable programs in the path.

I thought there could be more cases like this in external libraries, and in the interest of time and effort, to proceed with the sledgehammer approach of compiling to jar on the same platform as native compilation.

Let me know if you want me to investigate further and reproduce a general test case outside of the above observation. 

(cc'ing @borkdude for awareness)

Reply:
Would it help if babashka fs removed :const?

Reply:
I'll fix it now.

Reply:
Try `0.1.11`

Reply:
Thanks @borkdude 

I've updated `babashka.fs` to v0.1.11 and have tested it to work, the windows integration test now passes.

Just for the record, the "cross compilation" is still susceptible to this type of errors since we don't control what goes in the external libraries. I suppose the expectation here is that such errors will be either caught up by users or during testing (currently there is very limited set for windows with this PR though there is a plan with #1211 to align the windows test coverage with Linux).